wallet-new/server/Cardano/Wallet/API/V1/LegacyHandlers/Transactions.hs

```diff
@@ -118,6 +122,46 @@ estimateFees Payment{..} = do
     cAccountId <- migrate pmtSource
     utxo <- V0.getMoneySourceUtxo ws (V0.AccountMoneySource cAccountId)
     outputs <- V0.coinDistrToOutputs =<< mapM migrate pmtDestinations
-    fee <- V0.rewrapTxError "Cannot compute transaction fee" $
-        eitherToThrow =<< V0.runTxCreator policy (V0.computeTxFee pendingAddrs utxo outputs)
-    single <$> migrate fee
+    efee <- V0.runTxCreator policy (V0.computeTxFee pendingAddrs utxo outputs)
+    single <$> case efee of
+        Left txError ->
+            case txError of
+                V0.NotEnoughMoney coin -> do
+                    acct <- V0.getAccount cAccountId
```

Comment:
Possibly not relevant here as this is a "benign" endpoint, but this doesn't look fully  _data-race free_, if I'm not mistaken. You could potentially, in a very fast-paced environment, yield an incorrect calculation if, by the time you grab the account via `getAccount`, the utxo of that account has changed.

But look, you already grabbed the `utxo` for that `accountId` via `getMoneySourceUtxo`, so you should be able to compute this light-hearted if you pass the `Utxo` as input to `mkLowerBound` and fold over it to grab the total `Coin`s for that account.

If the semantic is the same I think it is, `getMoneySourceUtxo` should give you only the _account-relevant_ portion of the available Utxo.